index.bs

```diff
@@ -3283,34 +3283,28 @@ size of 1024, it attempts to fill the developer-supplied buffer, allowing full c
       },
 
       pull(controller) {
-        if (controller.byobRequest) {
-          const v = controller.byobRequest.view;
-
-          return fs.read(fd, v.buffer, v.byteOffset, v.byteLength, position).then(bytesRead => {
-            if (bytesRead === 0) {
-              return fs.close(fd).then(() => controller.close());
-            } else {
-              position += bytesRead;
-              controller.byobRequest.respond(bytesRead);
-            }
-          });
-        } else {
-          // The consumer is using the default reader.
-          const buffer = new ArrayBuffer(DEFAULT_CHUNK_SIZE);
-          return fs.read(fd, buffer, 0, DEFAULT_CHUNK_SIZE, position).then(bytesRead => {
-            if (bytesRead === 0) {
-              return fs.close(fd).then(() => controller.close());
-            } else {
-              position += bytesRead;
-              controller.enqueue(new Uint8Array(buffer, 0, bytesRead));
-            }
-          });
-        }
+        // Even when the consumer is using the default reader, the auto allocate feature allocate a buffer and pass it
+        // to us via byobRequest.
+
+        assert(controller.byobRequest !== undefined);
+
+        const v = controller.byobRequest.view;
+
+        return fs.read(fd, v.buffer, v.byteOffset, v.byteLength, position).then(bytesRead => {
+          if (bytesRead === 0) {
+            return fs.close(fd).then(() => controller.close());
+          } else {
+            position += bytesRead;
+            controller.byobRequest.respond(bytesRead);
+          }
+        });
       },
 
       cancel() {
         return fs.close(fd);
-      }
+      },
+
+      autoAllocateChunkSize: DEFAULT_CHUNK_SIZE
     });
```

Comment:
Oh, I see, that makes sense. I forgot that we were inside onreadable, not inside pull(). I don't think we need another auto-allocation case for this.

> We can also choose not to drain data from the kernel while there's no outstanding read requests. But it doesn't match the original concept of the example?

Right, I am assuming that the kernel buffer might overflow if we do that.